pkg/analytics/analytics.go

```diff
@@ -2,57 +2,114 @@ package analytics
 
 import (
 	"context"
+	"crypto/tls"
 	"fmt"
-	"time"
 
+	"github.com/Masterminds/semver"
 	"github.com/rs/zerolog/log"
 	"go.opentelemetry.io/otel/attribute"
-	"go.opentelemetry.io/otel/exporters/otlp/otlplog/otlploggrpc"
+	"go.opentelemetry.io/otel/exporters/otlp/otlplog/otlploghttp"
 	"go.opentelemetry.io/otel/exporters/stdout/stdoutlog"
 	otellog "go.opentelemetry.io/otel/log"
+	"go.opentelemetry.io/otel/log/global"
 	sdklog "go.opentelemetry.io/otel/sdk/log"
 	"go.opentelemetry.io/otel/sdk/resource"
 	semconv "go.opentelemetry.io/otel/semconv/v1.26.0"
 
 	"github.com/bacalhau-project/bacalhau/pkg/models"
+	"github.com/bacalhau-project/bacalhau/pkg/version"
 )
 
-type Recorder interface {
-	EmitEvent(ctx context.Context, event EventType, properties ...otellog.KeyValue)
-	EmitJobEvent(ctx context.Context, event EventType, j models.Job)
-	Stop(ctx context.Context) error
+const ProviderKey = "bacalhau-analytics"
+
+const (
+	NodeInstallationIDKey = "installation_id"
+	NodeInstanceIDKey     = "instance_id"
+	NodeIDKey             = "node_id"
+	NodeTypeKey           = "node_type"
+	NodeVersionKey        = "node_version"
+)
+
+type Config struct {
+	attributes   []attribute.KeyValue
+	otlpEndpoint string
+}
+
+type Option func(*Config)
+
+func WithEndpoint(endpoint string) Option {
+	return func(c *Config) {
+		c.otlpEndpoint = endpoint
+	}
+}
+
+func WithNodeNodeID(id string) Option {
+	return func(c *Config) {
+		c.attributes = append(c.attributes, attribute.String(NodeIDKey, id))
+	}
+}
+
+func WithNodeType(isRequester, isCompute bool) Option {
+	return func(c *Config) {
+		var typ string
+		if isRequester && isCompute {
+			typ = "hybrid"
+		} else if isRequester {
+			typ = "orchestrator"
+		} else if isCompute {
+			typ = "compute"
+		}
+		c.attributes = append(c.attributes, attribute.String(NodeTypeKey, typ))
+	}
 }
 
-var _ Recorder = (*LogRecorder)(nil)
+func WithInstallationID(id string) Option {
+	return func(c *Config) {
+		c.attributes = append(c.attributes, attribute.String(NodeInstallationIDKey, id))
+	}
+}
 
-type LogRecorder struct {
-	provider *sdklog.LoggerProvider
+func WithInstanceID(id string) Option {
+	return func(c *Config) {
+		c.attributes = append(c.attributes, attribute.String(NodeInstanceIDKey, id))
+	}
 }
 
-func New(ctx context.Context, opts ...Option) (*LogRecorder, error) {
+func WithVersion(bv *models.BuildVersionInfo) Option {
+	return func(c *Config) {
+		v, err := semver.NewVersion(bv.GitVersion)
+		if err != nil {
+			// use the version populated via the `ldflags` flag.
+			c.attributes = append(c.attributes, attribute.String(NodeVersionKey, version.GITVERSION))
+		} else {
+			c.attributes = append(c.attributes, attribute.String(NodeVersionKey, v.String()))
+		}
+	}
+}
+
+func SetupAnalyticsProvider(ctx context.Context, opts ...Option) error {
 	config := &Config{
-		otlpEndpoint: "localhost:4317", // Default endpoint
+		otlpEndpoint: "localhost:4318", // Default endpoint
```

Comment:
lets switch to `t.bacalhau.dev:4317` with todo to move to `t.bacalhau.org:4317`

Reply:
done.